### PR TITLE
chore(go): Refactor

### PR DESCRIPTION
**Pull request by candiduslynx**
2 enhancements:
1. Save a bit if space (empty struct is 0, pointer is 4-8 bytes)
2. Ease the init (no need to explicitly allocate transformers)

### CODE REVIEW

File: csv/transformer.go [hunk 1]
```diff
@@ -6,79 +6,81 @@ import (
...
 type ReverseTransformer struct {
-	defaultTransformer destination.DefaultReverseTransformer
+	*destination.DefaultReverseTransformer
 }
 
-func (t *ReverseTransformer) ReverseTransformValues(table *schema.Table, values []any) (schema.CQTypes, error) {
-	return t.defaultTransformer.ReverseTransformValues(table, values)
-}
+var _ interface {
+	ReverseTransformValues(table *schema.Table, values []any) (schema.CQTypes, error)
```
Comment (hermanschaaf) [inline csv/transformer.go:13]: Why do we need / want this interface declaration? I get that it's essentially asserting that the interface is implement at compile-time, and it makes sense to me for the `schema.CQTypeTransformer` case, but why the custom interface here?

**Comment by hermanschaaf** [comment]
What is the reason for the refactor? Why is the new version better?

**Review by hermanschaaf — APPROVED** [review]

**Review by bbernays — APPROVED** [review]
I am going to approve this but... In my opinion this change makes the code less readable/ easy to understand, so the tradeoff of readability vs saving a few bytes isn't really worth it... But will defer to the rest of the team

**Comment by erezrokah** [comment]
> Save a bit if space (empty struct is 0, pointer is 4-8 bytes)

I'm not sure this is a valid concern for us at the moment, unless we see this impacts the overall experience of the module. So maybe we should refactor this when we see the need?

> Ease the init (no need to explicitly allocate transformers)

This is mostly a style issue we don't enforce (i.e. explicit init vs implicit init), so I could go either way. This PR has a side effect that CSV transformers are now available to the JSON client and vice versa, instead of being `nil`. I don't see it as a big  though.

TLDR: I understand what this PR does and why, but I'm not sure the refactor is required

File: csv/transformer.go (hunk 1 above)
Comment (candiduslynx) [reply csv/transformer.go:13]: Just wanted to make sure it works fine statically.
Basically, we should either have this interface declared in sdk or just drop
